src/sync/sync-api.js

```diff
@@ -77,19 +78,29 @@ export class SyncApi extends TypedEmitter {
   /** @type {Map<import('protomux'), Set<Buffer>>} */
   #pendingDiscoveryKeys = new Map()
   #l
+  #blobDownloadFilter
 
   /**
    *
    * @param {object} opts
    * @param {import('../core-manager/index.js').CoreManager} opts.coreManager
    * @param {CoreOwnership} opts.coreOwnership
    * @param {import('../roles.js').Roles} opts.roles
+   * @param {import('../types.js').BlobFilter | null} opts.blobDownloadFilter
```

Comment:
This is waiting for #940 to land so we can use the NON_ARCHIVE_DEVICE_FILTER constant for initializing the class. Left it out to avoid creating a dependency on that PR for landing this.